pkg/network/node/vnids.go

```diff
@@ -106,8 +107,17 @@ func (vmap *nodeVNIDMap) WaitAndGetVNID(name string) (uint32, error) {
 	if err == nil {
 		return id, nil
 	} else {
+		// We may find netid when we check with api server but we will
+		// still treat this as an error if we don't find it in vnid map.
+		// So that we can imply insufficient timeout if we see many VnidNotFoundErrors.
 		VnidNotFoundErrors.Inc()
-		return 0, fmt.Errorf("failed to find netid for namespace: %s in vnid map", name)
+
+		netns, err := vmap.networkClient.Network().NetNamespaces().Get(name, metav1.GetOptions{})
+		if err != nil {
+			return 0, fmt.Errorf("failed to find netid for namespace: %s, %v", name, err)
+		}
+		glog.Warningf("Netid for namespace: %s exists but not found in vnid map", name)
+		return netns.NetID, nil
```

Comment:
we should probably store this in the map since there will probably be further queries for it and we don't want to block 5 seconds again each time